src/test/scala/org/broadinstitute/dsde/workbench/sam/util/DistributedLockSpec.scala

```diff
@@ -0,0 +1,63 @@
+package org.broadinstitute.dsde.workbench.sam.util
+
+import cats.effect.IO
+import cats.kernel.Eq
+import org.broadinstitute.dsde.workbench.google.GoogleFirestoreOpsInterpreters
+import org.broadinstitute.dsde.workbench.model.WorkbenchException
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.broadinstitute.dsde.workbench.sam.Generator.genLockPath
+import org.broadinstitute.dsde.workbench.sam.TestSupport
+import org.broadinstitute.dsde.workbench.sam.util.DistributedLock._
+import org.scalatest.{AsyncFlatSpec, Matchers}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+
+class DistributedLockSpec extends AsyncFlatSpec with Matchers with TestSupport{
```

Comment:
If this requires real google it can't be a unit test

Reply:
I wanted to test against google api though. cuz I realize a few of my assumptions weren't correct when writing out this test...where should I move this kind of test?

Reply:
Then that is an integration test. There should be 2 levels to the testing, is your code right and are you call google right. The former is the unit test which should have mock google. The latter is an integration test that should be written in /automation... ask @marctalbott :)